django/core/handlers/base.py

```diff
@@ -69,6 +75,11 @@ def load_middleware(self):
                 self._response_middleware.insert(0, mw_instance.process_response)
             if hasattr(mw_instance, 'process_exception'):
                 self._exception_middleware.insert(0, mw_instance.process_exception)
+            if callable(mw_instance):
+                self._middleware.insert(0, mw_instance)
+
+        if middleware_count == len(self._middleware):
```

Comment:
I don't understand how this check implies that all middleware are new-style. It seems that it would be true anytime no middleware raised `MiddlewareNotUsed`, which seems probably pretty common even with old-style middleware. What am I missing?